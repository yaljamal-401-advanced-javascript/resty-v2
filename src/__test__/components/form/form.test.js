import React from 'react';
import { mount } from 'enzyme';
import Form from '../../../components/form/form.js';

describe('<Form/> (Enzyme Test)', () => {
  it('is alive at application start', () => {
    let app = mount(<Form />);
    expect(app.find('#get').text()).toBe('GET');
    expect(app.find('#post').text()).toBe('POST');
    expect(app.find('#put').text()).toBe('PUT');
    expect(app.find('#delete').text()).toBe('DELETE');
  });

  it('stores the temporary state of the form', () => {
    let url = 'http://www.foo.com';
    let method = 'get';
    let app = mount(<Form />);
    app.find('input').simulate('change', { target: { value: url } });
    app.find('#get').simulate('click');
    expect(app.state('url')).toBe(url);
    expect(app.state('method')).toBe(method);
  });

  it('removes the temporary state of the form on submit', () => {
    let url = 'http://www.foo.com';
    let app = mount(<Form />);
    app.find('input').simulate('change', { target: { value: url } });
    app.find('#get').simulate('click');
    app.find('form').simulate('submit');
    expect(app.state('url')).toBe('');
    expect(app.state('method')).toBe('');
  });

  it('stores the final state of the form', () => {
    let url = 'http://www.foo.com';
    let method = 'get';
    let app = mount(<Form />);
    app.find('input').simulate('change', { target: { value: url } });
    app.find('#get').simulate('click');
    app.find('form').simulate('submit');
    expect(app.state('request').method).toBe(method);
    expect(app.state('request').url).toBe(url);
  });

  it('renders the users selections', () => {
    let url = 'http://www.foo.com';
    let method = 'get';
    let app = mount(<Form />);
    app.find('input').simulate('change', { target: { value: url } });
    app.find('#get').simulate('click');
    app.find('form').simulate('submit');
    expect(app.find('.method').text()).toBe(method);
    expect(app.find('.url').text()).toBe(url);
  });
});
