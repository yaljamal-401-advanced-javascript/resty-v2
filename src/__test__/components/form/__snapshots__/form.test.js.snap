// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form/> (Enzyme Test) SnapShot 1`] = `
Array [
  <form
    onSubmit={[Function]}
  >
    <label>
      <span>
        URL: 
      </span>
      <input
        name="url"
        onChange={[Function]}
        type="text"
      />
      <button
        type="submit"
      >
        GO!
      </button>
    </label>
    <label
      className="methods"
    >
      <span
        className="active"
        id="get"
        onClick={[Function]}
      >
        GET
      </span>
      <span
        className=""
        id="post"
        onClick={[Function]}
      >
        POST
      </span>
      <span
        className=""
        id="put"
        onClick={[Function]}
      >
        PUT
      </span>
      <span
        className=""
        id="delete"
        onClick={[Function]}
      >
        DELETE
      </span>
    </label>
  </form>,
  <section
    className="results"
  >
    <span
      className="method"
    />
    <span
      className="url"
    />
  </section>,
]
`;
